tests/python_client/testcases/test_utility.py

```diff
@@ -2309,3 +2390,1259 @@ def test_role_list_grants(self, host, port):
+        self.utility_wrap.role_grant("Global", "*", "CreateCollection")
+        self.utility_wrap.role_grant("Global", "*", "All")
+
+        g_list, _ = self.utility_wrap.role_list_grant("Global", "*")
+        assert len(g_list.groups) == 2
+        for g in g_list.groups:
+            assert g.object == "Global"
+            assert g.object_name == "*"
+            assert g.privilege in ["CreateCollection", "All"]
+            self.utility_wrap.role_revoke(g.object, g.object_name, g.privilege)
+        self.utility_wrap.role_drop()
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_verify_admin_role_privilege(self, host, port):
+        """
+        target: verify admin role privilege
+        method: create a new user, bind to admin role, crud collection
+        expected: verify success
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        self.utility_wrap.init_role("admin")
+        user = cf.gen_unique_str(prefix)
+        password = cf.gen_unique_str(prefix)
+        c_name = cf.gen_unique_str(prefix)
+        u, _ = self.utility_wrap.create_user(user=user, password=password)
+
+        self.utility_wrap.role_add_user(user)
+
+        self.connection_wrap.disconnect(alias=DefaultConfig.DEFAULT_USING)
+        self.connection_wrap.connect(host=host, port=port, user=user,
+                                     password=password, check_task=ct.CheckTasks.ccr)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data(ct.default_nb)
+        collection_w.insert(data=data)
+        collection_w.load()
+        assert collection_w.num_entities == ct.default_nb
+        collection_w.release()
+        collection_w.drop()
+
+    @pytest.mark.tags(CaseLabel.L3)
+    def test_verify_user_owns_public_role_privilege(self, host, port):
+        """
+        target: verify public role privilege
+        method: create a new user, verify public role privilege
+        expected: verify success
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        user = cf.gen_unique_str(prefix)
+        password = cf.gen_unique_str(prefix)
+        c_name = cf.gen_unique_str(prefix)
+        u, _ = self.utility_wrap.create_user(user=user, password=password)
+        collection_w = self.init_collection_wrap(name=c_name)
+        self.connection_wrap.disconnect(alias=DefaultConfig.DEFAULT_USING)
+        self.connection_wrap.connect(host=host, port=port, user=user,
+                                     password=password, check_task=ct.CheckTasks.ccr)
+
+        assert collection_w.description == ""
+        assert self.utility_wrap.has_collection(c_name)
+        assert not collection_w.has_index()[0]
```

Comment:
No check for unsupported privilege for public here.

Reply:
It will be checked in test_new_user_default_owns_public_role_permission line 3165 
I will  delete test_verify_user_owns_public_role_privilege this case